### ISSUE TITLE
"get_slide_region" size is flipped for "image.TiledImage"

### ISSUE THREAD

**Issue by adamvonpaternos** [herrmannlab/dicomslide#5]
To give context, I'm retrieving slide regions using both saved annotations (TID 1410 Planar ROI Measurements and Qualitative Evaluations) and manually curated rois (slide offset and bounding box size). To that extent, for regions with relatively the same size bounding box, the image regions seem accurate. However for regions with an oblong bounding box, the return image seems to be stretched in the opposite direction. Please see the annotation in our viewer ([slim viewer v0.12](https://github.com/ImagingDataCommons/slim/tree/master/src)) and the returned region.

<img width="961" alt="viewer_annotation" src="https://user-images.githubusercontent.com/24463494/221255886-cc70394b-e110-4fe1-ba1d-c38274cd8391.png">

![current_region](https://user-images.githubusercontent.com/24463494/221301226-a4b6d3a1-997a-4ce7-8a0b-c07efcdbd404.png)

I'm leveraging dicomslide.slide.Slide [get_slide_region](https://github.com/ImagingDataCommons/dicomslide/blob/055a7a07a0a3fabd7c6e11c015da324435367b7e/src/dicomslide/slide.py#L960-L968) and [get_slide_region_for_annotation](https://github.com/ImagingDataCommons/dicomslide/blob/055a7a07a0a3fabd7c6e11c015da324435367b7e/src/dicomslide/slide.py#L1031-L1041) which call [dicomslide.image.TiledImage.get_slide_region](https://github.com/ImagingDataCommons/dicomslide/blob/055a7a07a0a3fabd7c6e11c015da324435367b7e/src/dicomslide/image.py#L707-L714) under the hood. It seems that there is a [contradiction](https://github.com/ImagingDataCommons/dicomslide/blob/055a7a07a0a3fabd7c6e11c015da324435367b7e/src/dicomslide/image.py#L772-L773) in row and col assignment. 

I've made a fix locally to dicomslide, which resulted in the "correctly" returned region (see below). This simply flips the index when extracting from size and pixel spacing, but unsure if this is expected.

```python
        region_rows = int(np.ceil(size[0] / self._pixel_spacing[0]))
        region_cols = int(np.ceil(size[1] / self._pixel_spacing[1]))
```

![flipped_region](https://user-images.githubusercontent.com/24463494/221301275-ee8ad4d0-7338-4a39-9afc-43fc68b02103.png)

Any advice would be great if this is an annotation/coordinate system issue with the viewer or actually a bug in the lib.

**Comment by hackermd** [comment]
Thanks @adamvonpaternos for delving into this issue.

I think the code should be conceptually correct. Are we sure the annotations are correct? Could you kindly run the script with debug log level and provide the log output (removing an potential PHI such as UIDs)?

**Comment by hackermd** [comment]
@adamvonpaternos thanks for the additional details. Weirdly your call of `get_slide_region_for_annotation()` doesn't log anything from within the *dicomslide* library, but it should (see [here](https://github.com/ImagingDataCommons/dicomslide/blob/c9e7467000881c085f4200eebcfeb161231260ef/src/dicomslide/image.py#L750-L754)).

Hmm...I wonder whether the change you proposed would do the trick. I will need to think through the logic again. What complicates the assessment is that the function rotates the image region before returning it (see [here](https://github.com/ImagingDataCommons/dicomslide/blob/c9e7467000881c085f4200eebcfeb161231260ef/src/dicomslide/image.py#L831-L833)). Looking through the code, I noticed that all unit tests are currently performed for the simplest image orientation (see [here](https://github.com/ImagingDataCommons/dicomslide/blob/055a7a07a0a3fabd7c6e11c015da324435367b7e/tests/test_slide.py#L138)). It may be worth parametrizing the tests with different orientations and see what happens. That could help us get to the bottom of the problem.

**Comment by hackermd** [comment]
@adamvonpaternos I think I found several bugs and hopefully fixed them with #7. Could you kindly take a look the PR and check whether these changes allow you to access slide regions as expected?
